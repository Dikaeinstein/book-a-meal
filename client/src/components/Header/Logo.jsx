import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';

const Logo = ({ text, className }) => (
  <NavLink
    to="/"
    activeClassName="active"
  >
    <h1 className={className}>{text}</h1>
  </NavLink>
);

Logo.propTypes = {
  text: PropTypes.string.isRequired,
  className: PropTypes.string.isRequired,
};

export default Logo;
