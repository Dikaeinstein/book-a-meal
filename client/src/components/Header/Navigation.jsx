import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';
import { removeUser } from '../../helpers/persistUser';
import history from '../../helpers/history';
import { userSignOut } from '../../actions/userActions';

const ConnectedNavigation = ({ urls, signout }) => {
  const handleSignout = (event) => {
    event.preventDefault();
    signout();
    removeUser();
    history.push('/');
  };

  const urlList = urls.map(url => (
    url.link === 'signout'
      ? (
        <li key={url.id}>
          <NavLink
            to={`/${url.link}`}
            onClick={handleSignout}
          >
            {url.name}
          </NavLink>
        </li>
      )
      : (
        <li key={url.id}>
          <NavLink
            to={`/${url.link}`}
          >{url.name}
          </NavLink>
        </li>
      )));

  return (
    <nav className="col-3-4">
      <ul className="nav-container text-center">
        {urlList}
      </ul>
    </nav>
  );
};

const mapStateToProps = state => ({
  urls: state.urls,
});

const mapDispatchToProps = dispatch => ({
  signout() { dispatch(userSignOut()); },
});

ConnectedNavigation.propTypes = {
  urls: PropTypes.arrayOf(PropTypes.object).isRequired,
  signout: PropTypes.func.isRequired,
};

const Navigation = connect(
  mapStateToProps, mapDispatchToProps,
  null, { pure: false },
)(ConnectedNavigation);

export default Navigation;
