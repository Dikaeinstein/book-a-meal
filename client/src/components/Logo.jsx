import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { userSignOut } from '../actions/userActions';

const ConnectedLogo = ({ text, className, goHome }) => {
  const handleClick = () => {
    goHome();
  };
  return (
    <NavLink to="/" onClick={handleClick}>
      <h1 className={className}>{text}</h1>
    </NavLink>
  );
};

ConnectedLogo.propTypes = {
  text: PropTypes.string.isRequired,
  className: PropTypes.string.isRequired,
  goHome: PropTypes.func.isRequired,
};

const mapDispatchToProps = dispatch => ({
  goHome: () => dispatch(userSignOut()),
});

const Logo = connect(null, mapDispatchToProps)(ConnectedLogo);

export default Logo;
