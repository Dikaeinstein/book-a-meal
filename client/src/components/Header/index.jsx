import React from 'react';
import PropTypes from 'prop-types';
import { ToastContainer } from 'react-toastify';
import 'react-toastify/dist/ReactToastify.min.css';
import Logo from './Logo';
import Navigation from './Navigation';

const Header = ({ className }) => (
  <header className={className}>
    <ToastContainer />
    <Logo
      text="Book-A-Meal"
      className="col-1-4 logo text-left text-dark"
    />
    <div className="toggle-show">&#9776;</div>
    <Navigation />
  </header>
);

Header.propTypes = {
  className: PropTypes.string.isRequired,
};

export default Header;
