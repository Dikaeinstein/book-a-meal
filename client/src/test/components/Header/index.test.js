import React from 'react';
import { shallow } from 'enzyme';
import { ToastContainer } from 'react-toastify';
import ConnectedNavigation from '../../../components/Header/Navigation';
import Header from '../../../components/Header';
import Logo from '../../../components/Header/Logo';

let wrapper;

describe('<Header />', () => {
  beforeAll(() => {
    /* eslint function-paren-newline: 0 */
    wrapper = shallow(<Header className="test" />);
  });
  it('should render a hamburger', () => {
    expect(wrapper.find('.toggle-show').length).toEqual(1);
  });
  it('should render a "ToastContainer" component', () => {
    expect(wrapper.find(ToastContainer).length).toEqual(1);
  });
  it('should render a "Logo" component', () => {
    expect(wrapper.find(Logo).length).toEqual(1);
  });
  it('should render a "Navigation" component', () => {
    expect(wrapper.find(ConnectedNavigation).length).toEqual(1);
  });
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
