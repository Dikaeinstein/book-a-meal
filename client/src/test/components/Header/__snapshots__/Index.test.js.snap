// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should match snapshot 1`] = `
<header
  className="test"
>
  <ToastContainer
    autoClose={5000}
    bodyClassName={null}
    className={null}
    closeButton={
      <CloseButton
        ariaLabel="close"
      />
    }
    closeOnClick={true}
    draggable={true}
    draggablePercent={80}
    hideProgressBar={false}
    newestOnTop={false}
    pauseOnHover={true}
    pauseOnVisibilityChange={true}
    position="top-right"
    progressClassName={null}
    rtl={false}
    style={null}
    toastClassName={null}
    transition={[Function]}
  />
  <Logo
    className="col-1-4 logo text-left"
    text="Book-A-Meal"
  />
  <div
    className="toggle-show"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    tabIndex="0"
  >
    â˜°
  </div>
  <Connect(Component) />
</header>
`;
